data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt

```diff
@@ -0,0 +1,29 @@
+package io.gnosis.data.repositories
+
+import io.gnosis.data.backend.TransactionServiceApi
+import io.gnosis.data.backend.dto.tokenAsErc20Token
+import io.gnosis.data.db.daos.Erc20TokenDao
+import io.gnosis.data.models.Balance
+import io.gnosis.data.models.Erc20Token
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import java.math.BigInteger
+
+class TokenRepository(
+    private val erc20TokenDao: Erc20TokenDao,
+    private val transactionServiceApi: TransactionServiceApi
+) {
+
+    suspend fun loadBalanceOf(safe: Solidity.Address): List<Balance> =
+        transactionServiceApi.loadBalances(safe.asEthereumAddressChecksumString())
+            .map {
+                val token = it.tokenAsErc20Token()
+                erc20TokenDao.insertToken(token)
```

Comment:
Do we load the token anywhere? Or is this just for future caching?

Reply:
Future caching. But maybe it is pointless. It was basically to not let all the time I put into the migration tests and all that go to waste 😆 If I remove this, then I can't justify having an ERC20TokenDao and then the futility back propagates :D

Reply:
should we remove these db inserts for now? No reason to have them here after we agreed to always load fresh data when opening screens

Reply:
I would prefer still keeping it. Like I explained before, I would have to then remove the DAO, tests for it, migration tests, etc. I don't see any harm in caching this stuff, specially if eventually we are going to have to implement it anyway. I think the cost of re-writing this in the future far overweights the performance benefits for the user.